app/models/manageiq/providers/ansible_tower/automation_manager/job.rb

```diff
@@ -169,6 +169,18 @@ def raw_stdout(format = 'txt')
     raise MiqException::MiqOrchestrationStatusError, err.to_s, err.backtrace
   end
 
+  def raw_artifacts
+    ext_management_system.with_provider_connection do |connection|
+      connection.api.jobs.find(ems_ref).artifacts
+    end
```

Comment:
Future refactoring but most of these methods do "with_provider_connection { |c| c.api.jobs.find(ems_ref) }` which is essentially what `with_provider_object { |job| }` does.